IdentityCore/src/cache/mac/MSIDMacTokenCache.m

```diff
@@ -22,39 +22,448 @@
 // THE SOFTWARE.
 
 #import "MSIDMacTokenCache.h"
+#import "MSIDToken.h"
+#import "MSIDTokenCacheKey.h"
+#import "MSIDTokenSerializer.h"
+#import "MSIDUserInformation.h"
+
+#define CURRENT_WRAPPER_CACHE_VERSION 1.0
+
+#define RETURN_ERROR_IF_CONDITION_FALSE(_cond, _code, _details) { \
+    if (!(_cond)) { \
+        NSError* _MSID_ERROR = MSIDCreateError(MSIDErrorDomain, _code, _details, nil, nil, nil, nil, nil); \
+        if (error) { *error = _MSID_ERROR; } \
+        return NO; \
+    } \
+}
+
+@interface MSIDMacTokenCache ()
+
+@property (nonatomic) NSMutableDictionary *cache;
+@property (nonatomic) dispatch_queue_t synchronizationQueue;
+
+@end
 
 @implementation MSIDMacTokenCache
 
-- (BOOL)setItemWithKey:(MSIDTokenCacheKey *)key
-                  item:(MSIDToken *)item
-            serializer:(id<MSIDTokenSerializer>)serializer
-               context:(id<MSIDRequestContext>)context
-                 error:(NSError **)error
+- (dispatch_queue_t)synchronizationQueue
 {
-    return NO;
+    if (!_synchronizationQueue) {
+        NSString *queueName = [NSString stringWithFormat:@"com.microsoft.msidmactokencache-%@", [NSUUID UUID].UUIDString];
+        _synchronizationQueue = dispatch_queue_create([queueName cStringUsingEncoding:NSASCIIStringEncoding], DISPATCH_QUEUE_CONCURRENT);
+    }
+    
+    return _synchronizationQueue;
+}
+
++ (MSIDMacTokenCache *)defaultCache
+{
+    static dispatch_once_t once;
+    static MSIDMacTokenCache *cache = nil;
+    
+    dispatch_once(&once, ^{
+        cache = [MSIDMacTokenCache new];
+    });
+    
+    return cache;
+}
+
+- (nullable NSData *)serialize
+{
+    if (!self.cache)
+    {
+        return nil;
+    }
+    
+    __block NSDictionary *cacheCopy = nil;
+    dispatch_sync(self.synchronizationQueue, ^{
+        cacheCopy = [self.cache mutableCopy];
+    });
+    
+    NSData *result = nil;
+    
+    // Using the dictionary @{ key : value } syntax here causes _cache to leak. Yay legacy runtime!
```

Comment:
Any information about that :)?

Reply:
Added #57 to investigate that behaviour.